cmd/fossil/config.go

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (c) 2022, Gideon Williams <gideon@gideonw.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+package fossil
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"reflect"
+	"time"
+
+	"github.com/rs/zerolog"
+	"github.com/spf13/viper"
+)
+
+func initConfig(configFile string) {
+	log := viper.Get("logger").(zerolog.Logger)
+
+	// config Read
+	viper.SetConfigType("toml")
+	viper.AddConfigPath("config")
+	viper.AddConfigPath("/etc/fossil/")
+	viper.AddConfigPath("$HOME/.fossil")
+	viper.AddConfigPath(".")
+	viper.SetConfigFile(configFile)
+
+	err := viper.ReadInConfig()
+	if _, ok := err.(viper.ConfigFileNotFoundError); ok {
+		log.Info().Msg("No config file found, using defaults as a base")
+	} else if err != nil {
+		log.Error().Msg("Error loading config file")
+	}
+
+	databaseConfigs := viper.GetStringMap("database")
+	databases := []string{}
+
+	// Range over the database blocks to set a list of database names
+	for k, v := range databaseConfigs {
```

Comment:
So is the intention here that the server would create a single database object per configured database? i.e. on startup, the server would read this config and hold say 10 databases in memory?

Reply:
I am not sure how we would want to do it, but that is one possibility. It could be simple to do that.

Add a command for `SETDB` in the mux that changes the active db on that connection and then at the server level we would have a map of DBs to name.